sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -402,9 +402,78 @@ export class StreamingReceiver extends MessageReceiver {
     await this._receiverHelper.suspend();
   }
 
-  async init(useNewName: boolean, abortSignal?: AbortSignalLike): Promise<void> {
-    const options = this._createReceiverOptions(useNewName, this._getHandlers());
-    await this._init(options, abortSignal);
+  private static wrapRetryOperation(
+    entityPath: string,
+    fullyQualifiedNamespace: string,
+    fn: () => Promise<void>,
+    onError?: OnError
+  ) {
+    return async () => {
+      try {
+        await fn();
+      } catch (err) {
+        onError &&
+          onError({
+            error: err,
+            errorSource: "receive",
+            entityPath,
+            fullyQualifiedNamespace
+          });
+
+        if (err.name !== "AbortError") {
+          err.retryable = true;
+        }
+
+        // once we're at this point where we can spin up a connection we're past the point
+        // of fatal errors like the connection string just outright being malformed, for instance.
+        throw err;
+      }
+    };
+  }
+
+  async init(
+    args: { useNewName: boolean; connectionId: string; onError: OnError } & Pick<
+      OperationOptionsBase,
+      "abortSignal"
+    >
+  ) {
+    while (true) {
+      const config: RetryConfig<void> = {
+        operation: StreamingReceiver.wrapRetryOperation(
+          this.entityPath,
+          this._context.config.host,
+          () => this._initOnce(args),
+          args.onError
+        ),
+        connectionId: args.connectionId,
+        operationType: RetryOperationType.receiverLink,
+        // even though we're going to loop infinitely we allow them to control the pattern we use on each
+        // retry run. This lets them toggle things like exponential retries, etc..
```

Comment:
Do the options like exponential-retries make sense anymore?
If the retry operation fails after the retries are exhausted, we invoke retry again where the exponential retries are basically reset.

Reply:
I think it still makes sense in that it controls what we do within a retry cycle. There is some discontinuity between cycles but I think it's still a good balance. The user can always specify _more_ retries if they want.